### PR TITLE
feat(MAJOR): Major refactor of all modules and examples

### PR DESCRIPTION
**Pull request by acelebanski**
## Description

This PR introduces v3.0 of the repo and contains:

- major refactor to all modules (except for legacy `virtual_machine` module)
- replacement of `application_insights` module with a new `ngfw_metrics` module
- introduction of `test_infrastructure` module
- significant refactor to all reference architecture examples
- bump of required Terraform version to **1.5**
- bump of required AzureRM provider version to **3.98**

Details are described in the issues mentioned in the section below.

## Motivation and Context

Big part of the work has been done on the old archived repo, issue here:
https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/issues/307

Work has been finished on the new repo and documented in this issue:
https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/issues/3

## How Has This Been Tested?

All modules were thoroughly tested "on the go". At the end, all examples were manually tested end-to-end.

## Types of changes

- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by acelebanski** [comment]
/plan paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries"

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.5 1.6 1.7"

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries" tf_version="1.5 1.6 1.7"

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries" tf_version="1.5 1.6 1.7"

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.5 1.6 1.7"

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.5 1.6 1.7"

File: examples/common_vmseries_and_autoscale/example.tfvars [hunk 1]
```diff
@@ -178,28 +205,46 @@ appgws = {
...
+    }
+    authentication = {
+      disable_password_authentication = false
+    }
+    virtual_machine_scale_set = {
+      bootstrap_options = "type=dhcp-client"
+      zones             = ["1", "2", "3"]
+    }
+    autoscaling_configuration = {
+      default_count = 2
```
Comment (alperenkose) [inline examples/common_vmseries_and_autoscale/example.tfvars:241]: Shall we put a comment on here stating that this should match with the `default_profile` count in the `autoscaling_profiles` ?

Comment (acelebanski) [reply examples/common_vmseries_and_autoscale/example.tfvars:241]: Don't really think we should. Both properties are described well in the README and customer can configure all different kinds of autoscaling profiles, ours are just simple examples.

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.5 1.6 1.7"

**Comment by acelebanski** [comment]
/idempotence paths="examples/common_vmseries examples/common_vmseries_and_autoscale examples/dedicated_vmseries examples/dedicated_vmseries_and_autoscale examples/gwlb_with_vmseries examples/standalone_panorama examples/standalone_vmseries" tf_version="1.5 1.6 1.7"

**Review by michalbil — APPROVED** [review]
Great work! 🚀

**Pull request merged**